modules/ROOT/pages/_partials/partner-and-customer-guide.adoc

```diff
@@ -130,10 +132,29 @@ See the https://docs.oracle.com/en/java/javase/17/migrate/getting-started.html#G
 See xref:java-support.adoc#mule-runtime[Mule runtime] to learn more about the Java 17 changes made in Mule runtime engine. 
 +
 ** Reflection on Java classes
++
+To ensure compatibility with Java 17, review and update any code that uses reflection to access Java classes or set parameter values. Java 17 imposes stricter access controls, so you should avoid introspecting JDK internals. Modify your code to eliminate deep introspection of JDK classes and ensure that reflective access is limited to necessary areas only.
```

Comment:
`to necessary areas only.` -> This is ambiguos. Necessity is not the the guiding principle but what Java 17 would allow. We can provide high level guidelines  like use of getters and setters over reflective field access, use of explicit constructors, etc, but reality is that it doesn't make sense to try and list every possible issue and scenario, specially without asking the customer to become expert on Mule classloading.  Other than the basics, the best recommendation here is to have a robust test suite and make sure it  is run with Java 17